apis/python/src/tiledb/vector_search/index.py

```diff
@@ -35,11 +36,15 @@ class Index:
     timestamp: int or tuple(int)
         If int, open the index at a given timestamp.
         If tuple, open at the given start and end timestamps.
+    load_index_data: bool
```

Comment:
The functionality of `load_index_data` is overlapping a bit with `memory_budget` and it is not clear which to use for each usecase. Could we rename this to something more indicative of the functionality of this PR i.e. `open_for_remote_query_execution`?